File: 0101-symmetric-tree/0101-symmetric-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function (root) {
    if (root == null || (root.left == null && root.right == null)) return true
    let queue = [root]

    while (queue.length > 0) {
        left_arr = []
        right_arr = []
        let length = queue.length
        for (let i = 0; i < length; i++) {
            let first = queue.shift()
            if (first.left != null) {
                left_arr.push(first.left ? first.left.val : null)
                queue.push(first.left)
            }
            if (first.right != null) {
                right_arr.push(first.right ? first.right.val : null)
                queue.push(first.right)
            }
            if (first.right == null && first.left == null) {
                right_arr.push(null)
                left_arr.push(null)
            }
        }
        left_arr = left_arr.reverse().join(",")
        if (left_arr == right_arr.join(',')) {
            left_arr = ''
            right_arr = ''
        }
        else {
            return false
        }
    }
    return true
};

File: 0101-symmetric-tree/0101-symmetric-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isSymmetric(self, root: Optional[TreeNode]) -> bool:
        def again(root1, root2):
            if root1 == None and root2 == None:
                return True
            if root1 == None or root2 == None:
                return False
            if root1.val != root2.val:
                return False
            left = again(root1.left, root2.right)
            right = again(root1.right, root2.left)
            return left and right
        return again(root.left,root.right)
